packages/flutter/test/widgets/implicit_animations_test.dart

```diff
@@ -154,7 +174,30 @@ void main() {
 
   testWidgets('AnimatedOpacity onEnd callback test', (WidgetTester tester) async {
     await tester.pumpWidget(wrap(
-        child: TestAnimatedWidget(callback: mockOnEndFunction, switchKey: switchKey, state: _TestAnimatedOpacityWidgetState(),)
+      child: TestAnimatedWidget(
+        callback: mockOnEndFunction,
+        switchKey: switchKey,
+        state: _TestAnimatedOpacityWidgetState(),
+      )
+    ));
+
+    final Finder widgetFinder = find.byKey(switchKey);
+
+    await tester.tap(widgetFinder);
+
+    await tester.pump();
+    expect(mockOnEndFunction.called, 0);
+    await tester.pump(animationDuration);
```

Comment:
can you add a check when pump a certain duration (< animationDuration) and verify the RenderSliverAnimatedOpacity.opacity.value = some expected value

Reply:
Great idea! I'll add that to the box and sliver one!